### PR TITLE
Adds "captions" object to configure the styles of the captions (group labels)

### PR DESCRIPTION
**Pull request by stockiNail**
Creates new captions element to configure the styles of the captions (related to the groups) when the chart is grouping. 

Adds the following styling options:

  * **display**: boolean which defines if the captions must be shown.
  * **align**: 'left' | 'center' | 'right', defines the horizontal alignment of the captions, inside the element.
  * **color**: defines the color to apply to the captions.
  * **formatter**: defines a callback to invoke to customize to the caption label to show.
  * **font**: defines the font to apply to the captions.
  * **padding**: number which defines the amount of pixels to apply to the captions XY position.

Removes `groupLabels `option because the `captions.display` is used instead.
Removes `font` and `hoverFont` option because the `captions.font` and `captions.hoverFont` are used instead.
Removes `color` and `hoverColor` option because the `captions.color` and `captions.hoverColor` are used instead.

- [x] test cases
- [x] documentation

### CODE REVIEW

**Comment by kurkle** [comment]
Looks good, go ahead with the docs! 💯

**Comment by stockiNail** [comment]
@kurkle about scriptable options, I see that in JS code all options are scriptable but not in TS definition.

The doc reflects TS definition. Anyway I think some options could not be scriptable, like `padding` and `borderWidth` because needed in a part of the code, for instance `buildData` function, where the scriptable options are not resolved.

I think the descriptor should reflect the TS definition, doesn't it?

File: docs/usage.md [hunk 1]
```diff
@@ -89,27 +89,23 @@ These are used to set display properties for a specific dataset.
...
-| [`groupDividers`](#general) | `boolean` | Yes | `false` |
-| [`groupLabels`](#general) | `boolean` | Yes | `true` |
+| [`borderWidth`](#styling) | `number`\|`object` | - | `0`
+| [`captions`](#captions) | `object` | - | 
+| [`dividerCapStyle`](#divider) | `string` | - | `'butt'`
+| [`dividerColor`](#divider) | [`Color`](https://www.chartjs.org/docs/latest/general/colors.html) | - | `'black'`
+| [`dividerDash`](#divider) | `number[]` | - | `undefined`
+| [`dividerDashOffset`](#divider) | `number` | - | `0`
+| [`dividerWidth`](#divider) | `number` | - | `1`
+| [`groupDividers`](#general) | `boolean` | - | `false` |
```
Comment (kurkle) [inline docs/usage.md:99]: Just a note to self, should scope these under `dividers` :)

**Comment by kurkle** [comment]
> @kurkle about scriptable options, I see that in JS code all options are scriptable but not in TS definition.
> 
> The doc reflects TS definition. Anyway I think some options could not be scriptable, like `padding` and `borderWidth` because needed in a part of the code, for instance `buildData` function, where the scriptable options are not resolved.

You are correct, the `dataset` passed to `buildData` is a plain object, not a resolver, so anything used from there is not scriptable. (I think if `me.options` was passed instead, then everything would be scriptable (in dataset context))

> I think the descriptor should reflect the TS definition, doesn't it?

What do you mean by descriptor? If documentation, then yes :)

**Review by kurkle — APPROVED** [review]
Should add some tests for the scriptable options. But that's out of scope for this PR.

**Comment by stockiNail** [comment]
> What do you mean by descriptor? If documentation, then yes :)

No, I meant the descriptors (I forgot the "s"):

https://github.com/kurkle/chartjs-chart-treemap/blob/cb202d48f5b61c47c4491412dddc019e0c5bb1af/src/controller.js#L297-L300

**Comment by kurkle** [comment]
> > What do you mean by descriptor? If documentation, then yes :)
> 
> No, I meant the descriptors (I forgot the "s"):
> 
> https://github.com/kurkle/chartjs-chart-treemap/blob/cb202d48f5b61c47c4491412dddc019e0c5bb1af/src/controller.js#L297-L300

Right, there is not need to limit the descriptors, except for callbacks and the like that should not be resolved automatically.

File: docs/usage.md (hunk 1 above)
Comment (stockiNail) [reply docs/usage.md:99]: @kurkle  yes!! maybe you don't believe me but it is in my TODO list but if you want to do it, more than happy!!!

Comment (kurkle) [reply docs/usage.md:99]: I'll get to it eventually. Currently too busy with other stuff though. So I'll welcome a PR too :)

Comment (stockiNail) [reply docs/usage.md:99]: > I'll get to it eventually. Currently too busy with other stuff though. So I'll welcome a PR too :)

I'm busy too (I'm using my spare time) but I'll do my best to submit a PR, sooner or later.

Comment (stockiNail) [reply docs/usage.md:99]: > 
> 
> Just a note to self, should scope these under `dividers` :)

PR #55 ;)

**Pull request merged**